agent/api/task/task.go

```diff
@@ -2402,7 +2402,7 @@ func (task *Task) dockerPortMap(container *apicontainer.Container) (nat.PortMap,
 			// we will try to get a contiguous set of host ports from the ephemeral host port range.
 			// this is to ensure that docker maps host ports in a contiguous manner, and
 			// we are guaranteed to have the entire hostPortRange in a single network binding while sending this info to ECS.
-			hostPortRange, err := getHostPortRange(numberOfPorts, protocol)
+			hostPortRange, err := getHostPortRange(numberOfPorts, protocol, cfg.PortRange)
```

Comment:
this would change to `cfg.DynamicHostPortRange` based on comments above. applicable to all places.